airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py

```diff
@@ -0,0 +1,257 @@
+            params = dict(urllib.parse.parse_qsl(next_query_string))
+            return params
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        return {"Content-Type": "application/json"}
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        if response_json.get("error"):
+            raise Exception(repr(response_json.get("error")))
+        result = response_json.get(self.data_field, [])
+        yield from result
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return next_page_token or {}
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        with vcr.use_cassette(cache_file.name, record_mode="new_episodes", serializer="json"):
+            yield from super().read_records(
+                sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state
+            )
+
+
+class IncrementalSurveymonkeyStream(SurveymonkeyStream, ABC):
+
+    state_checkpoint_interval = 1000
+
+    @property
+    @abstractmethod
+    def cursor_field(self) -> str:
+        """
+        Defining a cursor field indicates that a stream is incremental, so any incremental stream must extend this class
+        and define a cursor field.
+        """
+
+    def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["sort_order"] = "ASC"
+        params["sort_by"] = "date_modified"
+        params["per_page"] = 1000  # maybe as user input or bigger value
+        since_value = pendulum.parse(stream_state.get(self.cursor_field)) if stream_state.get(self.cursor_field) else self._start_date
+
+        since_value = max(since_value, self._start_date)
+        params["start_modified_at"] = since_value.strftime("%Y-%m-%dT%H:%M:%S")
+        return params
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_state = latest_record.get(self.cursor_field)
+        current_state = current_stream_state.get(self.cursor_field) or latest_state
+        if current_state:
+            return {self.cursor_field: max(latest_state, current_state)}
+        return {}
+
+
+class Surveys(IncrementalSurveymonkeyStream):
```

Comment:
can you add a big block comment above this surveys stream that describes the strategy we are using to pull data from the surveymonkey API? Since we're not using the endpoints in a standard way (seems like we're hitting the same endpoint for everything) it's hard to follow exactly how this logic works.